png_benchmark.py

```diff
@@ -0,0 +1,131 @@
+                "max_size_B": 0,
+                "min_size_B": 0,
+                "avg_size_B": 0,
+                "max_compression_%": 0,  # % of original size
+                "min_compression_%": 0,
+                "avg_compression_%": 0,
+                "frame_data": []
+            } for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL+1)
+        } for compress_type in COMPRESS_TYPES
+    }
+
+    test_begin = time.time()
+    print("Start time:", test_begin)
+
+
+    for compress_type in COMPRESS_TYPES:
+        print(f"-----------------COMPRESS TYPE {compress_type}--------------------")
+        for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL+1):
+            max_time = 0
+            min_time = float("inf")
+            total_time = 0
+            max_size = 0
+            min_size = float("inf")
+            total_size = 0
+            max_compression = 0
+            min_compression = float("inf")
+            total_compression = 0
+            for frame in range(FRAME_COUNT):
+                img = Image.open(pathlib.Path(INPUT_PATH, f"{frame}.png"))
+                buffer = io.BytesIO()
+
+                # Running encode
+                gc.disable()
+                start = time.time_ns()
+                img.save(buffer, format="PNG", compress_level=compress_level, compress_type=compress_type)
+                end = time.time_ns()
+                gc.enable()
+
+                # Save singular test results
+                time_ns = end-start
+                size_B = buffer.getbuffer().nbytes
+                compression = size_B / os.path.getsize(pathlib.Path(INPUT_PATH, f"{frame}.png")) * 100
+
+                if time_ns > max_time:
+                    max_time = time_ns
+                elif time_ns < min_time:
+                    min_time = time_ns
+                
+                if size_B > max_size:
+                    max_size = size_B
+                elif size_B < min_size:
+                    min_size = size_B
+                
+                if compression > max_compression:
+                    max_compression = compression
+                elif compression < min_compression:
+                    min_compression = compression
+                
+                total_time += time_ns
+                total_size += size_B
+                total_compression += compression
+                test_result = {
+                    "time_ns": time_ns,
+                    "size_B": size_B,
+                    "compression_%": compression
+                }
+                results[f"compress_type_{compress_type}"][f"compress_level_{compress_level}"]["frame_data"].append(test_result)
+
+                # Save one image (this one has 2 landing pads in it) for reference
+                if frame == FRAME_TO_SAVE:
+                    img.save(f"{OUTPUT_PATH}/ct{compress_type}_cl{compress_level}.png", "PNG", compress_level=compress_level, compress_type=compress_type)
```

Comment:
Use `pathlib.Path` to create the path, then `str(image_path)` to get it to automatically format the directory separators for you.